util/flocks/src/tfrwlock/read_guard.rs

```diff
@@ -0,0 +1,23 @@
+use std::ops::Deref;
+use tokio::sync::RwLockReadGuard;
+use rustix::fd::AsFd;
+use crate::frwlock::{FrwLock, FrwLockReadGuard};
+
+pub struct TfrwLockReadGuard<'a, T: AsFd> {
+    _outer: RwLockReadGuard<'a, FrwLock<T>>,
+    inner: FrwLockReadGuard<T>,
+}
+
+impl<'a, T: AsFd> TfrwLockReadGuard<'a, T> {
+    pub(crate) fn new(outer: RwLockReadGuard<'a, FrwLock<T>>, inner: FrwLockReadGuard<T>) -> Self {
+        Self { _outer : outer, inner }
+    }
+}
+
+impl<'a, T: AsFd> Deref for TfrwLockReadGuard<'a, T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &*self.inner
+    }
+}
```

Comment:
We shouldn't be needing to roll our own read write guards. Is there no crate that gives us a file based `RwLock` ? I mean even a lightweight DB like reddis would be easier. We get everything out of the box with that.

Reply:
On Reddit, some had claimed this took over: https://github.com/alfiedotwtf/file-lock. But, it doesn't seem well maintained. Might as well eat our own toast. 

With Redis, you would then need to run the Redis service.

Reply:
Yes I meant Redis! How about using the Redis KV DB for a store?

Reply:
Also just found this, a transactional key-value store, pure rust implementation https://github.com/khonsulabs/nebari